### PR TITLE
Relay performance cleanup

### PR DESCRIPTION
**Pull request by Raynos**
This is a subset of the perf branch.

 - Remove double id protection to gain performance
 - Simplify code.
 - Re-use peer to gaurd against cannot req until identified edgecase

r: @kriskowal @shannili @rf

### CODE REVIEW

File: node/connection_base.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ function TChannelConnectionBase(channel, direction, socketRemoteAddr) {
         connection: self
     });
 
-    self.guid = ++CONNECTION_BASE_IDENTIFIER;
+    self.guid = String(++CONNECTION_BASE_IDENTIFIER + '~');
```
Comment (kriskowal) [inline node/connection_base.js]: The string cast is superfluous. Adding a string coerces implicitly.

Comment (Raynos) [inline node/connection_base.js]: True;

Comment (anson627) [inline node/connection_base.js]: ++CONNECTION_BASE_IDENTIFIER + '~' is already a string, why call String constructor again? Just out of curiosity.

**Comment by Raynos** [comment]
Fixed.

**Comment by ShanniLi** [comment]
lgtm

File: node/relay_request.js [hunk 2]
```diff
@@ -116,7 +142,7 @@ RelayRequest.prototype.createOutResponse = function createOutResponse(options) {
             remoteAddr: self.inreq.remoteAddr,
             id: self.inreq.id
         });
-        return;
+        return null;
```
Comment (anson627) [inline node/relay_request.js]: what's benefit of return null instead of undefined?

Comment (Raynos) [inline node/relay_request.js]: lint.

Comment (anson627) [inline node/relay_request.js]: make sense.

**Comment by kriskowal** [comment]
:ok:

**Pull request merged**